patterns/contracts-as-note-owners/escrow/src/test/e2e.test.ts

```diff
@@ -0,0 +1,273 @@
+            const txReceipt = escrow
+            .withWallet(alice)
+            .methods.escrow(alice.getAddress(), alice.getAddress(), FEE, token.address, nonce, recipients)
+            .simulate();
+
+            await expect(txReceipt).rejects.toThrow();
+        })
+
+        it("should mine the transaction", async () => {
+            const nonce = await createAuthEscrowMessage(
+                token,
+                alice,
+                escrow.address,
+                FEE
+            );
+            const txReceipt = await escrow
+            .withWallet(alice)
+            .methods.escrow(alice.getAddress(), alice.getAddress(), FEE, token.address, nonce, recipients)
+            .send()
+            .wait({debug: true});
+
+            const { visibleNotes } = txReceipt.debugInfo!;
+            notes = visibleNotes
+
+            expect(txReceipt.status).toBe("mined");
+        })
+        
+        it("should create 3 notes", async () => {
+            // Note1: EscrowNote: Alice
+            // Note2: ValueNote: Alice (due to initial mint)
+            // Note3: EscrowNote: Bob
+            expect(notes.length).toBe(3);
+        })
+
+        it("should create a note for alice with the correct parameters", async () => {
+            // Note1: EscrowNote: Alice
+            // Note2: ValueNote: Alice
+            // Note3: EscrowNote: Bob
+            const amountParam = notes[0].note.items[0];
+            const ownerParam = notes[0].note.items[1];
+            const tokenParam = notes[0].note.items[3];
+            const noteOwner = notes[0].owner;
+            
+            expect(amountParam.toBigInt()).toEqual(FEE);
+            expect(ownerParam).toEqual(alice.getAddress());
+            expect(tokenParam).toEqual(token.address);
+            expect(noteOwner).toEqual(alice.getAddress());
+
+            randomIdAlice = notes[0].note.items[2];
+        })
+
+        it("should create a note for bob with the correct parameters", async () => {
+            // Note1: EscrowNote: Alice
+            // Note2: ValueNote: Alice
+            // Note3: EscrowNote: Bob
+            const amountParam = notes[2].note.items[0];
+            const ownerParam = notes[2].note.items[1];
+            const tokenParam = notes[2].note.items[3];
+            const noteOwner = notes[2].owner;
+            
+            expect(amountParam.toBigInt()).toEqual(FEE);
+            expect(ownerParam).toEqual(alice.getAddress());
+            expect(tokenParam).toEqual(token.address);
+            expect(noteOwner).toEqual(bob.getAddress());
+
+            randomIdBob = notes[2].note.items[2];
+        })
+
+        it("should have the same randomId for both escrow notes", async () => {
+            expect(randomIdAlice).toEqual(randomIdBob);
+            randomId = randomIdAlice;
+        })
+
+        it("should decrease Alice's token balance", async () => {
```

Comment:
how is this one passing with the transfer commented out?

Reply:
oh, it's only the private balance while they're unshielded

Reply:
maybe we should have a helper for total (pub+priv) balance, that's an easy footgunning

Reply:
`unshield` moves funds from private balance to the of a chosen recipient, so the `transfer_public` wasn't needed. I just completely missed I left it out commented hehe.